modules/gapi/src/executor/gexecutor.cpp

```diff
@@ -291,6 +302,14 @@ class cv::gimpl::GExecutor::Output final: public cv::gimpl::GIslandExecutable::I
     {
         set(rcs);
     }
+
+    void verify()
+    {
+        if (eptr)
+        {
+            std::rethrow_exception(eptr);
```

Comment:
is it some kind of default/stub implementation?

because it now has a "state" and it affects multithreading behavior between simultaneous `post` and `verify` at least

Reply:
`GExecutor::IOutput` methods might be called from different threads, but not concurrently.
`verify()` is supposed to be called after `post()`

https://github.com/TolyaTalamanov/opencv/blob/at/handle-exception-in-streamingexecutor/modules/gapi/src/executor/gexecutor.cpp#L410-L412

That's a good point, but I don't know how guarantee that's methods won't be called concurrently in future, except putting comment